packages/evm/src/precompiles/wasm.ts

```diff
@@ -0,0 +1,168 @@
+import { ethers, InterfaceAbi } from 'ethers';
+import { Abi } from 'viem';
+/**
+ * Represents the functions available in the WASM precompile contract,
+ * facilitating interoperability between the EVM and Cosmos.
+ * @category Cosmos Interoperability
+ */
+export interface WasmPrecompileFunctions {
+	/**
+	 * Executes a message on the specified contract with provided coins.
+	 * @param contractAddress The address of the contract to execute the message on.
+	 * @param msg The message to execute.
+	 * @param coins The coins to provide for execution.
+	 * @returns A Promise resolving to an object containing the response.
+	 * @category Cosmos Interoperability
+	 */
+	execute(contractAddress: string, msg: Uint8Array, coins: Uint8Array): Promise<{ response: Uint8Array }>;
```

Comment:
i think the abi for this changed and coins isn't passed in as an argument anymore - the contract execute is a payable function so u just pass funds in directly